client.go

```diff
@@ -1375,3 +1397,545 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
+	}
+	wrappedAK, err := e3dbClients.EncryptAccessKey(keyResp, encryptionKeys)
+	if err != nil {
+		return err
+	}
+	accessKeyReq := storageClient.GroupAccessKeyRequest{
+		GroupID:            group.GroupID,
+		RecordType:         recordType,
+		EncryptedAccessKey: wrappedAK,
+		PublicKey:          group.PublicKey,
+	}
+	_, encryptedGroupAK, err := c.StorageClient.CreateGroupAccessKey(ctx, accessKeyReq)
+	if err != nil {
+		return err
+	}
+	// share secret with group
+	secretShareReq := storageClient.ShareGroupRecordRequest{
+		GroupID:            group.GroupID,
+		RecordType:         recordType,
+		EncryptedAccessKey: encryptedGroupAK,
+		PublicKey:          group.PublicKey,
+	}
+	_, err = c.StorageClient.ShareRecordWithGroup(ctx, secretShareReq)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetOrCreateNamespace creates the group for the namespace if it doesn't exist and returns the Group
+func (c *ToznySDKV3) GetOrCreateNamespace(ctx context.Context, realmName string, namespace string) (*storageClient.Group, error) {
+	var group *storageClient.Group
+	groupName := fmt.Sprintf("tozny.secret.%s.%s", realmName, namespace)
+	listRequest := storageClient.ListGroupsRequest{
+		ClientID:   uuid.MustParse(c.StorageClient.ClientID),
+		GroupNames: []string{groupName},
+	}
+	responseList, err := c.StorageClient.ListGroups(ctx, listRequest)
+	if err != nil {
+		return nil, err
+	}
+	encryptionKeyPair, err := e3dbClients.GenerateKeyPair()
+	if err != nil {
+		return nil, err
+	}
+	eak, err := e3dbClients.EncryptPrivateKey(encryptionKeyPair.Private, c.StorageClient.EncryptionKeys)
+	if err != nil {
+		return nil, err
+	}
+	if len(responseList.Groups) < 1 {
+		// create group
+		groupReq := storageClient.CreateGroupRequest{
+			Name:              groupName,
+			PublicKey:         encryptionKeyPair.Public.Material,
+			EncryptedGroupKey: eak,
+		}
+		createGroupResponse, err := c.StorageClient.CreateGroup(ctx, groupReq)
+		if err != nil {
+			return nil, err
+		}
+		group = createGroupResponse
+		// get membership keys
+		membershipKeyRequest := storageClient.CreateMembershipKeyRequest{
+			GroupAdminID:      c.StorageClient.ClientID,
+			NewMemberID:       c.StorageClient.ClientID,
+			EncryptedGroupKey: createGroupResponse.EncryptedGroupKey,
+			ShareePublicKey:   c.StorageClient.EncryptionKeys.Public.Material,
+		}
+		membershipKeyResp, err := c.StorageClient.CreateGroupMembershipKey(ctx, membershipKeyRequest)
+		if err != nil {
+			return nil, err
+		}
+		// make the capabilities
+		groupMemberCapabilities := []string{storageClient.ShareContentGroupCapability, storageClient.ReadContentGroupCapability}
+		// make a new member
+		memberReq := []storageClient.GroupMember{}
+		memberReq = append(memberReq, storageClient.GroupMember{
+			ClientID:        uuid.MustParse(c.StorageClient.ClientID),
+			MembershipKey:   membershipKeyResp,
+			CapabilityNames: groupMemberCapabilities,
+		})
+		// add group members
```

Comment:
as an example of future changes, I can imagine wanting to past in some members that should be added to the group by default besides myself